### PR TITLE
refactor: extract video events

### PR DESCRIPTION
**Pull request by csiyang**
# Description

- Extracted out video events to own component
- Removed the use of video play state
- Added additional mutations for START, PLAY, PAUSE, COMPLETED events

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/26787374/todos/4816010650)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [x] Tests pass

# Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged into main

### CODE REVIEW

**Comment by csiyang** [comment]
Requires [this](https://github.com/JesusFilm/core/pull/603) branch to be merged first

File: libs/journeys/ui/src/components/Video/VideoEvents/VideoEvents.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+          }
+        }
+      })
+    })
+
+    player.on('ended', () => {
+      void videoCompleteEventCreate({
+        variables: {
+          input: {
+            id: uuidv4(),
```
Comment (tataihono) [inline libs/journeys/ui/src/components/Video/VideoEvents/VideoEvents.tsx:102]:
```suggestion
            id: uuidv4(),
```

don't care about these. Its not required.

**Review by tataihono — CHANGES_REQUESTED** [review]
remove id's from the client side on input. Don't need them.

File: libs/journeys/ui/src/components/Video/VideoEvents/VideoEvents.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    VIDEO_COMPLETE_EVENT_CREATE
+  )
+
+  useEffect(() => {
+    player.on('ready', () => {
+      void videoStartEventCreate({
+        variables: {
+          input: {
+            blockId,
+            position: Math.floor(Math.floor(player.currentTime()))
```
Comment (tataihono) [inline libs/journeys/ui/src/components/Video/VideoEvents/VideoEvents.tsx:66]: why two math floor here?

File: libs/journeys/ui/src/components/Video/VideoEvents/VideoEvents.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+    render(
+      <MockedProvider
+        mocks={[
+          startMock,
+          {
+            request: {
+              query: VIDEO_COMPLETE_EVENT_CREATE,
+              variables: {
+                input: { blockId: 'video0.id', position: 0 }
```
Comment (tataihono) [inline libs/journeys/ui/src/components/Video/VideoEvents/VideoEvents.spec.tsx:178]: any way to test a non zero position? play.setCurrentTime(50.5) or something just so we know its working

**Review by tataihono — CHANGES_REQUESTED** [review]

File: libs/journeys/ui/src/components/Video/VideoEvents/VideoEvents.tsx (hunk 2 above)
Comment (csiyang) [reply libs/journeys/ui/src/components/Video/VideoEvents/VideoEvents.tsx:66]: oops

File: libs/journeys/ui/src/components/Video/VideoEvents/VideoEvents.spec.tsx (hunk 3 above)
Comment (csiyang) [reply libs/journeys/ui/src/components/Video/VideoEvents/VideoEvents.spec.tsx:178]: It has to be an int, but will update the test

**Review by tataihono — CHANGES_REQUESTED** [review]

**Pull request merged**